test_runner/fixtures/types.py

```diff
@@ -0,0 +1,109 @@
+import random
+from functools import total_ordering
+from typing import Union
+
+
+@total_ordering
+class Lsn:
+    """
+    Datatype for an LSN. Internally it is a 64-bit integer, but the string
+    representation is like "1/123abcd". See also pg_lsn datatype in Postgres
+    """
+
+    def __init__(self, x: Union[int, str]):
```

Comment:
I tried that now, but it didn't feel like an improvement. I'd like to keep the short `Lsn(str)` contsructor for string, as it's very frequently used. The integer variant is much less frequently used, so it might make sense to have a separate 'from_int(cls, x: int)` for that, but to construct the object in `from_int`, you'd need to call the constructor that takes string. So it would end up looking something like:

```

    def __init__(self, x: str):
        """Convert lsn from hex notation to int."""
        l, r = x.split("/")
        self.lsn_int = (int(l, 16) << 32) + int(r, 16)

    @classmethod
    def from_int(cls, x: int):
        return cls(f"{:X}/{:X}".format(x >> 32, x & 0xFFFFFFFF)
```
That seems a bit silly. You could work around that, with an optional 'int' argument to __init__ or some other tricks, but it won't look pretty.

mypy is pretty good at catching typing mixups, so I'm not very worried about confusing the string and integer variants.

Reply:
Makes sense, I didnt think about dominating frequency of Lsn(str) calls. Thanks!